src/dataflow/src/source/postgres.rs

```diff
@@ -92,7 +115,7 @@ impl PostgresSourceReader {
             .await?;
 
         let slot_query = format!(
-            r#"CREATE_REPLICATION_SLOT {:?} TEMPORARY LOGICAL "pgoutput" USE_SNAPSHOT"#,
+            r#"CREATE_REPLICATION_SLOT {:?} LOGICAL "pgoutput" USE_SNAPSHOT"#,
```

Comment:
would it be helpful to keep `TEMPORARY` until we're sure we clean up upstream slots?

Reply:
Unfortunately we need to drop the temporary keyword here because we need the slot to persist across connections in order to correctly resume replication. If we created a new temporary slot on every reconnection we'd risk having the WAL segments we need to be cleaned up (since no slot claims them during the offline period)